### PR TITLE
enhance: optimize retrieve on dynamic field

### PR DESCRIPTION
**Pull request by zhagnlu**
#35514

### CODE REVIEW

File: internal/core/src/segcore/SegmentInterface.cpp [hunk 1]
```diff
@@ -56,10 +56,21 @@ SegmentInternalInterface::FillTargetEntry(const query::Plan* plan,
...
 
+    std::unique_ptr<DataArray> field_data;
     // fill other entries except primary key by result_offset
     for (auto field_id : plan->target_entries_) {
-        auto field_data =
-            bulk_subscript(field_id, results.seg_offsets_.data(), size);
+        if (plan->schema_.get_dynamic_field_id().has_value() &&
+            plan->schema_.get_dynamic_field_id().value() == field_id &&
+            !plan->target_dynamic_fields_.empty()) {
+            auto target_dynamic_fields = plan->target_dynamic_fields_;
```
Comment (czs007) [inline internal/core/src/segcore/SegmentInterface.cpp:65]: will this copy the vector?

File: internal/core/src/segcore/SegmentInterface.cpp [hunk 2]
```diff
@@ -167,6 +178,16 @@ SegmentInternalInterface::FillTargetEntry(
             continue;
         }
 
+        if (plan->schema_.get_dynamic_field_id().has_value() &&
+            plan->schema_.get_dynamic_field_id().value() == field_id &&
+            !plan->target_dynamic_fields_.empty()) {
+            auto target_dynamic_fields = plan->target_dynamic_fields_;
```
Comment (czs007) [inline internal/core/src/segcore/SegmentInterface.cpp:184]: Ditto

File: internal/core/src/segcore/SegmentInterface.cpp (hunk 1 above)
Comment (zhagnlu) [reply internal/core/src/segcore/SegmentInterface.cpp:65]: ok, will auto& instead

**Pull request merged**